bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -551,8 +554,13 @@ private static SnapshotDescription toSnapshotDesscription(Snapshot snapshot) {
   }
 
   @Override
-  public CompletableFuture<Void> modifyTable(TableDescriptor arg0) {
-    throw new UnsupportedOperationException("modifyTable"); // TODO
+  public CompletableFuture<Void> modifyTable(TableDescriptor newDescriptor) {
+    return getDescriptor(newDescriptor.getTableName()).thenApply(descriptor -> tableModificationAdapter
+        .buildModifications(new HTableDescriptor(newDescriptor), new HTableDescriptor(descriptor)))
+        .thenApply(modifications -> {
+          modifyColumns(newDescriptor.getTableName(), modifications);
```

Comment:
`modifyColumns` returns a `CompletableFuture`, which means that we have to do a `join()` or `get()`, to ensure that the top CompletableFuture only returns with the modification is complete.